// kaggle-node/dist/constants/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.constants = void 0;
const DatasetQueryEnums_1 = require("../enums/DatasetQueryEnums");
exports.constants = {
    baseUrl: 'https://www.kaggle.com',
    apiPath: "api/v1",
    timeouts: {
        connect: 5000,
        read: 15000
    },
    envVarNames: {
        cache: "KAGGLEHUB_CACHE",
        baseUrl: "KAGGLE_API_ENDPOINT",
        username: "KAGGLE_USERNAME",
        key: "KAGGLE_KEY",
        usernameJson: "username",
        keyJson: "key",
        credFolder: "KAGGLE_CONFIG_DIR",
        verbosity: "KAGGLEHUB_VERBOSITY",
        disableCache: "DISABLE_KAGGLE_CACHE",
        disableColabCache: "DISABLE_COLAB_CACHE",
        runtimeAddr: "TBE_RUNTIME_ADDR"
    },
    filtering: {
        sortBy: Object.values(DatasetQueryEnums_1.DatasetQuerySorting),
        fileTypes: Object.values(DatasetQueryEnums_1.DatasetQueryFileTypes),
        licenses: Object.values(DatasetQueryEnums_1.DatasetQueryLicenses),
    },
    defaults: {
        datasetQueryParams: {
            'sortBy': DatasetQueryEnums_1.DatasetQuerySorting.HOTTEST,
            'filetype': DatasetQueryEnums_1.DatasetQueryFileTypes.ALL,
            'license': DatasetQueryEnums_1.DatasetQueryLicenses.ALL,
            'tagids': '',
            'search': '',
            'user': '',
            'page': 1,
        }
    }
};
