// kaggle-node/dist/classes/KaggleNodeClient.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KaggleNodeClient = void 0;
const axios_1 = __importDefault(require("axios"));
const constants_1 = require("./../constants/constants");
class KaggleNodeClient {
    constructor(creds, config) {
        var _a;
        this.axiosInstance = axios_1.default.create({
            baseURL: `${constants_1.constants.baseUrl}/${constants_1.constants.apiPath}`,
            auth: creds != null
                ? { username: creds === null || creds === void 0 ? void 0 : creds.username, password: creds === null || creds === void 0 ? void 0 : creds.key }
                : undefined,
            timeout: (_a = config === null || config === void 0 ? void 0 : config.timeout) !== null && _a !== void 0 ? _a : 10000,
        });
    }
    get(path, config) {
        return this.axiosInstance(Object.assign({ method: 'get', url: path }, (config !== null && config !== void 0 ? config : {})));
    }
    post(path, data, config) {
        return this.axiosInstance(Object.assign({ method: 'post', url: path, data: data }, (config !== null && config !== void 0 ? config : {})));
    }
}
exports.KaggleNodeClient = KaggleNodeClient;
