// kaggle-node/dist/enums/DatasetQueryEnums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatasetQueryGroups = exports.DatasetQueryLicenses = exports.DatasetQueryFileTypes = exports.DatasetQuerySorting = void 0;
var DatasetQuerySorting;
(function (DatasetQuerySorting) {
    DatasetQuerySorting["HOTTEST"] = "hottest";
    DatasetQuerySorting["VOTES"] = "votes";
    DatasetQuerySorting["UPDATED"] = "updated";
    DatasetQuerySorting["ACTIVE"] = "active";
    DatasetQuerySorting["PUBLISHED"] = "published";
})(DatasetQuerySorting || (exports.DatasetQuerySorting = DatasetQuerySorting = {}));
var DatasetQueryFileTypes;
(function (DatasetQueryFileTypes) {
    DatasetQueryFileTypes["ALL"] = "all";
    DatasetQueryFileTypes["CSV"] = "csv";
    DatasetQueryFileTypes["SQLITE"] = "sqlite";
    DatasetQueryFileTypes["JSON"] = "json";
    DatasetQueryFileTypes["BIGQUERY"] = "bigQuery";
})(DatasetQueryFileTypes || (exports.DatasetQueryFileTypes = DatasetQueryFileTypes = {}));
var DatasetQueryLicenses;
(function (DatasetQueryLicenses) {
    DatasetQueryLicenses["ALL"] = "all";
    DatasetQueryLicenses["CC"] = "cc";
    DatasetQueryLicenses["GPL"] = "gpl";
    DatasetQueryLicenses["ODB"] = "odb";
    DatasetQueryLicenses["OTHER"] = "other";
})(DatasetQueryLicenses || (exports.DatasetQueryLicenses = DatasetQueryLicenses = {}));
var DatasetQueryGroups;
(function (DatasetQueryGroups) {
    DatasetQueryGroups["PUBLIC"] = "public";
    DatasetQueryGroups["USER"] = "user";
})(DatasetQueryGroups || (exports.DatasetQueryGroups = DatasetQueryGroups = {}));
